resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -697,6 +700,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         executorTerminated(d, slaveId, taskId, s"Executor finished with state $state")
         // In case we'd rejected everything before but have now lost a node
         metricsSource.recordRevive
+        logInfo("Reviving offers due to a finished executor task.")
```

Comment:
This log message looks a little bit misleading. Do we address the situation with a failed node specifically? From the message body, it looks like we start reviving offers because one of the tasks completed successfully.

Reply:
That's right, we start reviving offers when one of the tasks is completed successfully as well when it fails. It's how it was implemented, I didn't change this behavior, just added a log message.

Reply:
We should only revive if we are going to use offers to launch a task/executor. For spark that sounds like when executor fails but not when it succeeds? If yes, let's fix that as part of this effort. Not sure if this is the right place to revive, is there another place in the code where intent to launch is recorded? Also, note that revive can get dropped, so this should be done in some kind of periodic manner as long as there are executors/tasks to be launched.